src/node-to-fsa/__tests__/NodeFileSystemDirectoryHandle.test.ts

```diff
@@ -0,0 +1,424 @@
+import { DirectoryJSON, memfs } from '../..';
+import { NodeFileSystemDirectoryHandle } from '../NodeFileSystemDirectoryHandle';
+import { NodeFileSystemFileHandle } from '../NodeFileSystemFileHandle';
+import { NodeFileSystemHandle } from '../NodeFileSystemHandle';
+import { maybe } from './util';
+
+const setup = (json: DirectoryJSON = {}) => {
+  const fs = memfs(json, '/');
+  const dir = new NodeFileSystemDirectoryHandle(fs as any, '/');
+  return { dir, fs };
+};
+
+maybe('NodeFileSystemDirectoryHandle', () => {
+  test('can instantiate', () => {
+    const { dir } = setup();
+    expect(dir).toBeInstanceOf(NodeFileSystemDirectoryHandle);
+  });
+
+  describe('.keys()', () => {
+    test('returns an empty iterator for an empty directory', async () => {
+      const { dir } = setup();
+      const keys = dir.keys();
+      expect(await keys.next()).toEqual({ done: true, value: undefined });
+    });
+
+    test('returns a folder', async () => {
+      const { dir } = setup({ folder: null });
+      const list: string[] = [];
+      for await (const key of dir.keys()) list.push(key);
+      expect(list).toEqual(['folder']);
+    });
+
+    test('returns two folders', async () => {
+      const { dir } = setup({
+        folder: null,
+        'another/folder': null,
+      });
+      const list: string[] = [];
+      for await (const key of dir.keys()) list.push(key);
+      expect(list.length).toBe(2);
+    });
+
+    test('returns a file', async () => {
+      const { dir } = setup({
+        'file.txt': 'Hello, world!',
+      });
+      const list: string[] = [];
+      for await (const key of dir.keys()) list.push(key);
+      expect(list).toEqual(['file.txt']);
+    });
+  });
+
+  describe('.entries()', () => {
+    test('returns an empty iterator for an empty directory', async () => {
+      const { dir } = setup();
+      const keys = dir.entries();
+      expect(await keys.next()).toEqual({ done: true, value: undefined });
```

Comment:
nit: using `await expect(...).resolves` gives some nicer output if the test fails:

```suggestion
      await expect(keys.next()).resolves.toStrictEqual({ done: true, value: undefined });
```

Reply:
Yes, I know, but I don't like to rely too much on library specific (Jest in this case) assertions if that can be easily done with just code.